### ISSUE TITLE
As an editor i would like an iframe component to host events and third party content

### ISSUE THREAD

**Issue by nicholas-haggerty** [equinor/energyvision#425]
I frame attributes from current set up:

Link title which is for accessibility only (not visible in UI)
Link placement
Option to add cookie policy NONE / MARKETING / STATISTICS
Option to have different aspect ratios (16:9 - 1:1)
Option to specify height by pixel

![iframe equinor.JPG](https://images.zenhubusercontent.com/5fb7bd9b956d0a1c1ed319e1/418d6656-b3c4-42c1-94e1-af593653e4bc)

**Comment by SvSven** [comment]
Cookie policy to be added later pending #138

**Comment by SvSven** [comment]
Couple of questions we should consider:

1. Where will this component be used? I'm assuming it will be for news articles and events at least; will it also be used on topic pages?
3. Will it be a "standalone" component, or something we use in a rich text editor?
2. If we use it as a standalone component, will we need/want a background color like the other components?
3. Will it need a visible title in a heading above the iframe like other components?

**Comment by SvSven** [comment]
Answers to previous questions from @nicholas-haggerty:

1.	Yes all templates (news/topic/event). Priority would be topic and then event’s although design work still needed on that template before handover. 
2.	Yes. Stand-alone for topics. For news and events, it’s a slightly different set up for editors but I also think stand-alone would be best here (with perhaps dedicated placement on the template)
3.	Yes colour background options as other components.
4.	I think it would be good to have an optional title yes but in general I think we have an issue on certain components that if an editor doesn’t fill in a field (eg text component over line) the spacing remains for it. My thought is that if a field isn’t filled in, the spacing allocated shouldn’t be applied. Is that possible? If so, yes good idea to add a title field.

**Comment by SvSven** [comment]
This should now be ready for review @nicholas-haggerty @BirteThornquist 

Added a test iframe to:
- https://energyvision.app.radix.equinor.com/careers
- https://energyvision.app.radix.equinor.com/news/this-is-a-test

The one field that is missing is the ability to set a height. Not sure if we really need this one, as I've made it so that the aspect ratio is a required field but has the initial value of 16:9 - the height is then automatically calculated based on the width and aspect ratio in the web.

For aspect ratio options I added: 16:9, 4:3, and 1:1 - but we can adjust that if we need more/less.

The `Frame title` field is a required field if the editor has put in a URL as having a title on the iframe element is a requirement for accessibility.

![image.png](https://images.zenhubusercontent.com/5dba9d22e2edb600015fe6c0/369b731e-1676-4aee-93b2-08ae11d42f34)

**Comment by nicholas-haggerty** [comment]
@SvSven looking good. In terms of the height aspect. For a use case: we have elements like the share feed which is via an Iframe. Because an editor can spec the height, it means we don't have the scroll bar within the frame. https://www.equinor.com/en/investors/the-equinor-share.html
I tried the same component on the test topic page. It displays with the scroll.

**Comment by SvSven** [comment]
Ah yeah, that's a good point. We could add it as a pixel value I suppose - which I guess is how it's set up in AEM? But I don't think it's ever possible to get it 100% right; if I resize my browser down on that page I also end up with scroll bars inside the iframe.

**Comment by nicholas-haggerty** [comment]
![iframe pixel aem.JPG](https://images.zenhubusercontent.com/5fb7bd9b956d0a1c1ed319e1/1b2671fe-78cb-496f-92bf-971717e1688a)
Yup pixel value in AEM. Agree its not perfect at every break but solved adequately enough for most. Quite rare that we would need to use this option but think we should include.

**Comment by SvSven** [comment]
Have added a field where you can set a height in pixels now 👍

**Comment by nicholas-haggerty** [comment]
Great stuff @SvSven thanks for fixing this so fast. No more comments from me @BirteThornquist take a look and if ok we can close.

**Comment by BirteThornquist** [comment]
Looks good to me @SvSven @nicholas-haggerty.
